### PR TITLE
feat(ci): add CodeQL workflows

### PR DESCRIPTION
**Pull request by GImbrailo**
This PR adds the CodeQL workflow to enable code scanning on a scheduled basis as well as for pull requests

### Testing plan

image
*Example of issues found by CodeQL*

image
*Example of CodeQL finding no issues after fix*

https://github.com/ecadlabs/taqueria/security/code-scanning?query=pr%3A518+tool%3ACodeQL

image
*Searchable history of branch issues*

### CODE REVIEW

**Review by BGMurphy — APPROVED** [review]
LGTM

File: taqueria-protocol/taqueria-protocol-types.ts [hunk 1]
```diff
@@ -82,7 +82,7 @@ const commandType: unique symbol = Symbol()
 export class Command extends StringLike {
     [commandType]: void
     static create(value: string) : Command | undefined {
-        if (value.match(/^([A-Za-z-_ ]+ ?)((\[.+\] ?)|(\<.+\>) ?)*$/)) {
+        if (value.match(/^([A-Za-z-_ ]+ ?)((\[[^\]\[\n]*\] ?)|(<[^<>\n]+>) ?)*$/)) {
```
Comment (danielelisi) [inline taqueria-protocol/taqueria-protocol-types.ts:85]: is this an unrelated change?

Comment (GImbrailo) [reply taqueria-protocol/taqueria-protocol-types.ts:85]: Yes, this change was added to show what happens when an issue that CodeQL finds is fixed. Removing this change will cause the workflow to fail which I think is acceptable. There are also some other options to dismiss the CodeQL findings that are shown here:

image

Comment (GImbrailo) [reply taqueria-protocol/taqueria-protocol-types.ts]: I removed the change

**Review by danielelisi — APPROVED** [review]
